src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -382,37 +384,30 @@ public static byte[] UrlEncodeToBytes(byte[] value, int offset, int count)
                 return null;
             }
 
-            int cSpaces = 0;
-            int cUnsafe = 0;
+            bool foundSpaces = false;
+            int unsafeCount = 0;
 
             // count them first
             for (int i = 0; i < count; i++)
             {
                 char ch = (char)value[offset + i];
 
                 if (ch == ' ')
-                    cSpaces++;
+                    foundSpaces = true;
                 else if (!IsUrlSafeChar(ch))
-                    cUnsafe++;
+                    unsafeCount++;
             }
 
             // nothing to expand?
-            if (cSpaces == 0 && cUnsafe == 0)
+            if (!foundSpaces && unsafeCount == 0)
```

Comment:
Curious: can we make a similar change to #7988 in UrlEncodeToBytes and UrlDecodeToBytes - is there a more performant way to replace all occurences of one byte with another?

Maybe something like
```
if (unsafeCount == 0)
{
    var subarray = new byte[count];
    Buffer.BlockCopy(value, offset, subarray, 0, count);
    if (foundSpaces)
    {
        for (int i = 0; i < subarray.Length; i++) 
        {
            if (subarray[i] == (byte)' ')
            {
                subarray[i] = (byte)'+';
            }
        }
    }
}
```